src/modals/cpu-pinning/CpuPinningDataProvider.js

```diff
@@ -83,47 +111,93 @@ const getPlacementPolicyHostsIds = (vm) => {
   return hostsIds
 }
 
-const getCpus = (host) => {
-  const topology = host.cpu?.topology
-  if (topology) {
-    return topology.sockets * topology.cores * topology.threads
-  }
-  return 0
-}
-
-const mapVmToModel = (vm) => {
-  return {
-    id: vm.id,
-    name: vm.name,
-    cpuTopology: {
-      sockets: parseInt(getSockets(vm)),
-      cores: parseInt(getCores(vm)),
-      threads: parseInt(getThreads(vm)),
-    },
-    cpuPinningPolicy: vm.cpu_pinning_policy,
-    cpuPinnings: getPinnings(vm),
+const mapVmToPinnedEntity = (vm) => {
+  const pinned = new PinnedEntity(vm.id, vm.name)
+
+  const sockets = +getSockets(vm)
```

Comment:
can we move parsing integer to `getSockets()` and skip conversion here and below?

Reply:
yes, we can!